pkg/devfile/validate/generic/errors.go

```diff
@@ -0,0 +1,136 @@
+package generic
+
+import "fmt"
+
+// InvalidEventError returns an error if the devfile event type has invalid events
+type InvalidEventError struct {
+	eventType string
+	errorMsg  string
+}
+
+func (e *InvalidEventError) Error() string {
+	return fmt.Sprintf("%s type events are invalid: %s", e.eventType, e.errorMsg)
+}
+
+// DuplicateCommandError returns an error if the command is duplicate
+type DuplicateCommandError struct {
+	commandId string
+}
+
+func (e *DuplicateCommandError) Error() string {
+	return fmt.Sprintf("devfile has duplicate command IDs %q", e.commandId)
+}
+
+// ExecCommandMissingComponentError returns an error if the exec command does not have a component
+type ExecCommandMissingComponentError struct {
+	commandId string
+}
+
+func (e *ExecCommandMissingComponentError) Error() string {
+	return fmt.Sprintf("exec command %q must reference a component", e.commandId)
+}
+
+// ExecCommandMissingCommandLineError returns an error if the exec command does not have a command line
+type ExecCommandMissingCommandLineError struct {
+	commandId string
+}
+
+func (e *ExecCommandMissingCommandLineError) Error() string {
+	return fmt.Sprintf("exec command %q must have a command", e.commandId)
+}
+
+// InvalidCommandError returns an error if the command is an invalid type
+type InvalidCommandError struct {
+	commandId   string
+	commandType string
+}
+
+func (e *InvalidCommandError) Error() string {
+	return fmt.Sprintf("the command %q should be of type %s", e.commandId, e.commandType)
+}
+
+// ExecCommandInvalidContainerError returns an error if the exec command references an invalid container component
+type ExecCommandInvalidContainerError struct {
+	commandId string
+}
+
+func (e *ExecCommandInvalidContainerError) Error() string {
+	return fmt.Sprintf("the command %q does not map to a container component", e.commandId)
+}
+
+// CompositeDirectReferenceError returns an error if the composite command directly references itself
+type CompositeDirectReferenceError struct {
+	commandId string
+}
+
+func (e *CompositeDirectReferenceError) Error() string {
+	return fmt.Sprintf("the composite command %q cannot reference itself", e.commandId)
+}
+
+// CompositeIndirectReferenceError returns an error if the composite command indirectly references itself
+type CompositeIndirectReferenceError struct {
+	commandId string
+}
+
+func (e *CompositeIndirectReferenceError) Error() string {
+	return fmt.Sprintf("the composite command %q cannot indirectly reference itself", e.commandId)
+}
+
```

Comment:
I think the errors could me refactored more here, like instead of having separate error for each validation fail it could be `InvalidCommand`  `MissingInformation`